.github/workflows/ci-cd.yml

```
@@ -89,12 +89,81 @@ jobs:
+    if: github.ref == 'refs/heads/main'
+    permissions:
+      contents: write
+    outputs:
+      new_tag_created: ${{ steps.check_tag.outputs.NEW_TAG_CREATED }}
+      tag: ${{ steps.check_tag.outputs.TAG }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
+      
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)